### PR TITLE
Learner 3845 - Restructure Test Automation Project to separate iOS & Android Cases

### PR DESCRIPTION
**Pull request by nedo**
- Restructure Test Automation Project to separate iOS & Android Cases
- added return types in docstrings
- LEARNER-3847 - Implement logging framework to record executions in Test Automation Project
- removed /test/conftest.py - merged its contents in main file(conftest.py)

### CODE REVIEW

**Comment by nedo** [comment]
@raeeschachar can you please review this PR ?

**Comment by nedo** [comment]
@raeeschachar PR is ready for another pass

File: android/pages/android_new_logistration.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        """
+        Get Login Button
+
+        Returns:
+            webdriver element: Login Button element
+        """
+
+        button_login = self.driver.find_element_by_id(self.elements.new_logistration_sign_in_button)
+        return self.global_contents.validate_element(
+            button_login, button_login.text,
```
Comment (raeeschachar) [inline android/pages/android_new_logistration.py]: We don't need to give `button_login.text` here. 
In validate element function, you can write button_login.text. I am commenting on that function too for further clarification.

File: common/globals.py [hunk 2]
```diff
@@ -20,152 +20,98 @@ class Globals:
...
-        self.and_platform = Strings.ANDROID
-        self.and_platform_version = '8.0'
-        self.and_device_name = 'Nexus 6P'
-        self.app_package_name = 'org.edx.mobile'
+        self.ios_platform_version = '10.3'
+        self.android_device_name = 'Nexus 6P'
+        self.android_platform_version = '8.0'
+        self.project_log = project_log
 
     def validate_element(self, target_element, element_value, expected_value, error_msg):
```
Comment (raeeschachar) [inline common/globals.py]: We don't need to give another argument `element_value` here. `target_element.text` would do the same.

File: android/pages/android_new_logistration.py (hunk 1 above)
Comment (nedo) [reply android/pages/android_new_logistration.py]: I am going to replace that function all together soon.

File: android/tests/test_android_login.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        Verify following contents are visible on screen, 
+            "Back icon", "Sign In" Title, "User name or e-mail address" label, User Name edit-field
+            Password edit-field, "Forgot your password?" option, "Sign In" button,
+            "Or sing in with" label, "Facebook" button, "Google" button,
+            "By signing in to this app, you agree to the" label ,
+            "edX Terms of Service and Honor Code" option
+        Verify all screen contents have their default values
+
+        """
+        log = setup_logging
```
[the remark this answers is not in the document]
Comment (nedo) [reply android/tests/test_android_login.py]: This is the basic flow of test automation framework, I'll explain that in tomorrow's meet up

File: common/globals.py (hunk 2 above)
Comment (nedo) [reply common/globals.py]: I am going to replace that function all together soon.

**Comment by nedo** [comment]
@raeeschachar PR is ready for another pass

File: ios/tests/test_ios_whats_new.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
+"""
+    Whats New Test Module
+"""
+
+from input_data import InputData
+from common import strings
+from ios.pages.ios_whats_new import IosWhatsNew
+
+
+class TestIosWhatsNew:
```
Comment (raeeschachar) [inline ios/tests/test_ios_whats_new.py]: Are we going to inherit anything in the future?

File: ios/pages/ios_whats_new.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    New Logistrtion Page Module
+"""
+
+from time import sleep
+from common.globals import Globals
+from common import strings
+from ios.pages.ios_elements import IosElements
+
+
+class IosWhatsNew:
```
Comment (raeeschachar) [inline ios/pages/ios_whats_new.py]: Inheritance?

File: android/pages/android_elements.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+"""
+
+
+class AndroidElements:
+    """
+    Contains list elements names and their paths, accessible in Pages
+    """
+
+    def __init__(self, project_log):
+        self.log  = project_log
```
Comment (raeeschachar) [inline android/pages/android_elements.py]: we can add all these as variables, not sure if we need to add __init__ or class. Let me know if you think other way round

File: android/pages/android_main_dashboard.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            str: Login screen Activity Name
+         """
+
+        self.textview_drawer_account_option.click()
+        sleep(self.global_contents.medium_timeout)
+        textview_logout = self.driver.find_element_by_id(self.elements.account_logout_option)
+        textview_logout.click()
+        sleep(self.global_contents.medium_timeout)
+        self.log.info(self.driver.current_activity)
+        return self.driver.current_activity
```
Comment (raeeschachar) [inline android/pages/android_main_dashboard.py]: empty line, end of file :)

File: ios/tests/test_ios_whats_new.py (hunk 4 above)
Comment (nedo) [reply ios/tests/test_ios_whats_new.py]: yap, I've this point in mind for future, as project will grow.

**Comment by nedo** [comment]
@raeeschachar PR is ready for another pass

File: ios/tests/test_ios_whats_new.py (hunk 4 above)
Comment (nedo) [reply ios/tests/test_ios_whats_new.py]: Yap I've this point in mind, I think it will be needed in future as project will grow

**Comment by raeeschachar** [comment]
:+1: feel free to merge it for now from code perspective. I am not sure about mobile testing architecture tho. 
After removal of methods that you mentioned, we can go through all of the code again. 
Nice to have: pep8 and pylint runs on PRs. You can discuss it with @MirzaMubasharBaig

**Comment by nedo** [comment]
Thanks @raeeschachar, next PR will not have global function. And integrating Pylint and Pep8 is also in pipeline

**Comment by nedo** [comment]
@staubina can you please review this PR ?

**Comment by staubina** [comment]
@nedo  I have looked through the restructure and I think it is a great move forward.  It also looks like @raeeschachar did a very thorough Python review.  I believe that we can merge this PR and move forward from here.  This Repo will grow and change as we iterate on it. 

Please Squash your commits before merging.  

Important step from here:

- We need to get PyLint in place to save you time.  I will check in with @edx/testeng  to see how we add it to the Repo.

**Comment by jzoldak** [comment]
@staubina from a high level, you need to hook this repo up to travis, and tell travis what (if anything) to install and what to run. 

Breaking it down:
* put in a IT support ticket to hook the repo up to travis
* add a .travis.yml file that tells travis what to do
 * a good simple model might be https://github.com/edx/testeng-ci/blob/master/.travis.yml
 * for linting, we generally use the [edx-lint module](https://github.com/edx/edx-lint) to maintain ruleset consistency across edX repos

**Comment by staubina** [comment]
Thank you @jzoldak I will look into these steps soon.

**Pull request merged**